### DISCUSSION TITLE
Can Blink Shell display images in the terminal?

### DISCUSSION THREAD

**Discussion by carloscabanero** [blinksh/blink#1524] [Q&A] [answered]
What is the fastest way to view images from the terminal? How do I make my terminal display images? How do I view an image inside the terminal, without downloading it?

**Comment by carloscabanero** [comment, marked as answer]
Blink Shell supports displaying images over SSH when using [iTerm2 imgcat](https://iterm2.com/utilities/imgcat).

Download it to your bin folder on the remote server, add chmod +x, and pass the image you want to display.

```shell
wget https://iterm2.com/utilities/imgcat -P ~/.bin
chmod +x .bin/imgcat
imgcat image.png
```

> This assumes you already have a .bin folder in your `$PATH`.

![IMG_2368.png](https://res.craft.do/user/full/0ef94637-30d5-80f2-9fa1-0089b763dcf6/CF7DACA1-88E2-4C6C-B0C3-AD80868DCFD1_2/hC2yYvZZ8Nae6M2q1RhVyBkg1WwUqynEE6uYpQ0EZrAz/IMG_2368.png)

To display in terminal multiplexers or mosh, you'll need to use one of many command-line applications to view images. For example, [VIU](https://github.com/atanunq/viu):

![IMG_2371.png](https://res.craft.do/user/full/0ef94637-30d5-80f2-9fa1-0089b763dcf6/42B10E09-8F92-418D-A2AB-16BBA54B3DA1_2/MJ8lLbX2Mjk8N33474MZGYuckDxF8z6tQ3FyMK3joPgz/IMG_2371.png)

### More information

- VIU: [https://github.com/atanunq/viu](https://github.com/atanunq/viu)
- ImgCat: [https://github.com/danielgatis/imgcat](https://github.com/danielgatis/imgcat)
- iTerm2 imgcat: [https://iterm2.com/utilities/imgcat](https://iterm2.com/utilities/imgcat)

**Reply by Gex2501** [reply]
It would seem this is no longer the case, it did work with an old version of the script but now it doesn't display anything.

**Reply by carloscabanero** [reply]
We haven't made any changes there so this should work all the same. Are you running it inside Mosh or any multiplexer that may be modifying the output?

**Reply by Gex2501** [reply]
> We haven't made any changes there so this should work all the same. Are you running it inside Mosh or any multiplexer that may be modifying the output?

Nope, I'm using blink to ssh direct to my Mac. Here is a screenshot from my iPad as well as a ZIP with the two scripts.

![IMG_1076](https://github.com/user-attachments/assets/b7825d61-9e22-4547-bb00-3e61ffb9711e)
[imgcat.zip](https://github.com/user-attachments/files/18509936/imgcat.zip)

**Reply by blueridanus** [reply]
Doesn't work for me either. Tested with imgcat and viu, both on ssh and mosh.

**Reply by carloscabanero** [reply]
The changes we made to the terminal shouldn't have any effect on this. Does the older script that Gex attached work for you? In any case, making a note to self here to review this before Blink 26 release.

**Reply by plockn** [reply]
It wasn’t woeking for me on ipad,  despite not giving any erros,  the command ran but no output… it started working when i used the -l (small L) parameter -> imgcat -l file.jpg

**Reply by tykim9999** [reply]
same issue
